flink-runtime/src/main/java/org/apache/flink/streaming/api/checkpoint/CheckpointedFunction.java

```diff
@@ -169,4 +169,15 @@ public interface CheckpointedFunction {
      * @throws Exception Thrown, if state could not be created ot restored.
      */
     void initializeState(FunctionInitializationContext context) throws Exception;
+
+    /**
+     * This method is called when a snapshot for a checkpoint is requested. Execution of this method
+     * does not block the main thread, is performed in an asynchronous thread pool.
+     *
+     * @param context the context for drawing a snapshot of the operator
```

Comment:
I am not sure what `drawing` means here. Maybe simplify to  `the context from which a snapshot of the operator is created`